src/Schema/Directives/LikeDirective.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Nuwave\Lighthouse\Support\Contracts\ArgBuilderDirective;
+
+class LikeDirective extends BaseDirective implements ArgBuilderDirective
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+enum LikePercentageLocation {
+    START
+    END
+    BOTH
+}
+
+"""
+Use the client given value to add a `LIKE` conditional to a database query.
+"""
+directive @like(
+  """
+  Specify the database column to compare.
+  Only required if database column has a different name than the attribute in your schema.
+  """
+  key: String
+
+  """
+  Specify the positions of the % in the LIKE comparison. The default is BOTH.
+  """
+  percentage: LikePercentageLocation
```

Comment:
We can have maximum flexibility by making `percentageLocation` optional.

```suggestion
  """
  Fixate the positions of the % wildcards in the LIKE comparison.
  If specified, wildcard characters in the client-given input are escaped.
  If not specified, the client can pass wildcards unescaped.
  """
  percentageLocation: LikePercentageLocation
```